Library/Homebrew/cmd/desc.rb

```diff
@@ -47,10 +47,6 @@ def run
         end
 
         if search_type.present?
-          if !args.eval_all? && !Homebrew::EnvConfig.eval_all?
-            raise UsageError, "`brew desc --search` needs `--eval-all` passed or `HOMEBREW_EVAL_ALL` set!"
-          end
```

Comment:
Suggestion: We still probably want to keep this error message when `HOMEBREW_NO_INSTALL_FROM_API` is set. It's possible that no API JSON file exists or that it's outdated which would cause unexpected results. This is admittedly a rare case though.

Reply:
You can use `Homebrew::EnvConfig.no_install_from_api?` to check that environment variable.

Reply:
done.